panml/core/llm/huggingface.py

```diff
@@ -143,9 +140,99 @@ def predict(self, text: str, max_length: int=50, skip_special_tokens: bool=True,
             output_context['text'] = self.tokenizer.decode(output[0], skip_special_tokens=skip_special_tokens)
         output_context['text'] = output_context['text'].replace('\n', '')
         output_context['text'] = output_context['text'].strip()
-        
+
         return output_context
     
+    # Generate text
+    def predict(self, text: Union[str, list[str], pd.Series], max_length: int=50, skip_special_tokens: bool=True, 
+                display_probability: bool=False, num_return_sequences: int=1, temperature: float=0.8, 
+                top_p: float=0.8, top_k: int=0, no_repeat_ngram_size: int=3, 
+                prompt_modifier: list[dict[str, str]]=[{'prepend': '', 'append': ''}], keep_history: bool=False) -> dict[str, str]:
```

Comment:
What's the purpose of defining the `prompt_modifier` as a list of dictionaries? Do we sequentially go through the list and prepend and/or append the prompts to the data column?

Reply:
okay so the list of strings itself remains the same but just prepended and appended with different prompt modifiers? I thought it may have made more sense to have the returned output of a model generation be fed in as input with new prepends and appends?

Reply:
yeah it's working like you said, the returned output of the model is then added on with prepends and appends, then fed back in for next round. Each prepend and append is specified by the respective element in the prompt modifier list.

Reply:
Ahh yeah just saw your comment from above.